### DISCUSSION TITLE
WebClient에서 MultiValueMap만 사용해야 되는 상황

### DISCUSSION THREAD

**Discussion by Dae-Hwa** [Sc0up/scoup-backend#153] [Troubleshooting docs]
그냥 아래와 같은 형식의 DTO로 요청을 보낼 경우 `MediaType.APPLICATION_JSON`는 괜찮은데`MediaType.APPLICATION_FORM_URLENCODED`에 대한 valid한 request로 인식이 안되더라구요! 그래서 MultiValueMap타입으로 새로 만들었어요
```java
public class TokenRequest {

    private final String clientId;

    private final String clientSecret;

    private final String redirectUri;

    private final String code;
}
```

_Originally posted by @janeljs in https://github.com/Sc0up/scoup-backend/pull/84#discussion_r693983518_

**Comment by Dae-Hwa** [comment]
이래저래 쓸 일이 생길 것 같은데 따로 유틸을 만드는것도 괜찮을 것 같아요

**Reply by janeljs** [reply]
이래저래 쓸 일이 생길 것 같다는 거 어떤 경우를 생각하시는지 부연 설명 해주실 수 있을까요?ㅎㅎ

**Reply by Dae-Hwa** [reply]
MultiValueMap이 사용되는 이유가 Request 주고받을 때 일반적인 Map으로 주고받으면 리스트의 경우를 보장할 수 가 없어서 만들어 놓은거라고 해요. 그래서 쓰이는건데,
마찬가지로 RestAssured같은데서 테스트용 파라미터를 감쌀때도 래핑해서 한꺼번에 넣으려면 MultiValueMap을 사용해야 합니다. 그런데 만들기가 엄청 귀찮아서 ㅎㅎ 그냥 스트링 값으로 된 key를 넣어서 리퀘스트를 만들게 되더라고요

https://jojoldu.tistory.com/478 참고했습니다

**Reply by janeljs** [reply]
DTO에서 MultiValueMap으로 변환하는 유틸클래스를 만드는 걸 말씀하신 거군요!ㅎㅎ 링크해주신 글 좋네요ㅎㅎ 적용해서 pr드릴게요😁
